ops/testing.py

```diff
@@ -1911,6 +1911,39 @@ def run_action(self, action_name: str,
                 output=action_under_test.output)
         return action_under_test.output
 
+    def set_cloud_spec(self, spec: 'model.CloudSpec'):
+        """Set cloud specification (metadata) including credentials.
+
+        Call this method before trying to call :meth:`ops.Model.get_cloud_spec`.
+
+        Example usage::
+
+            class MyVMCharm(ops.CharmBase):
+                ...
+                def _on_start(self, event: ops.StartEvent):
+                    spec = self.model.get_cloud_spec()
+                    ....
```

Comment:
How about we remove the `...` here, as I think it's very close to a fully working example. If we say `self.spec = self.model.get_cloud_spec()`, then we can `assert self.harness.charm.spec == expected` in the test, to make it a bit more realistic.